backend/src/v5/models/revisions.js

```diff
@@ -45,4 +50,21 @@ Revisions.getRevisionCount = (teamspace, model) => {
 	return db.count(teamspace, collectionName(model), query);
 };
 
+Revisions.getRevisions = async (teamspace, model, showVoid, projection = {}) => {
+	const query = { ...excludeIncomplete };
+
+	if (!showVoid) {
+		query.void = excludeVoids.void;
+	}
+
+	const revisions = await findRevisionsByQuery(teamspace, model, query, projection);
+	return revisions;
+};
+
+Revisions.updateRevisionStatus = async (teamspace, model, revision, status) => {
+	const query = { $or: [{ _id: revision }, { tag: revision }] };
+	await findOneRevisionByQuery(teamspace, model, query, { _id: 1, void: 1 });
```

Comment:
you don't need to run this query.

If you need to know if the revision existed, just run the `updateOne` query and [check the counts](https://docs.mongodb.com/manual/reference/method/db.collection.updateOne/#returns)